### ISSUE TITLE
QKSMS keeps crashing when trying to create a new message in a new conversation

### ISSUE THREAD

**Issue by mikoladz** [moezbhatti/qksms#1417]
### DESCRIPTION
Application QKSMS stops working when user is trying to create a new message in a new conversation (plus "+" icon).

### STEPS
1. Open QKSMS so that a list of conversations/messages is visible.
2. Tap "+" button to start a new conversation.
3. Application crashes with an error message "QKSMS has stopped" with an option to "Open app again".
4. Repeating the steps above causes another error message "QKSMS keeps stopping" with options either to "Open app again" or "Close app".

### EXPECTED
A list of contacts and their numbers appears where a user can type a new number or choose contact to send messages to.

### OBSERVATIONS
When "Mobile numbers only" is disabled in settings the problem above does not occur. It only occurs when the option is enabled.

QKSMS version 3.6.3
Device: Sony G8441
OS: Android 8.1.0/Lineage OS 15.1

**Comment by sphakka** [comment]
Exactly the same with QKSMS v3.6.4 on FP2 LinegeOS 15.1!

Relevant logcat lines:

```
05-11 12:12:05.576 19498 19518 I Finsky  : [411] iey.run(3): Stats for Executor: BlockingExecutor igj@becd478[Running, pool size = 0, active threads = 0, queued tasks = 0, completed tasks = 14]
05-11 12:12:05.587  5765  5765 D AndroidRuntime: Shutting down VM
05-11 12:12:05.589  5765  5765 E AndroidRuntime: FATAL EXCEPTION: main
05-11 12:12:05.589  5765  5765 E AndroidRuntime: Process: com.moez.QKSMS, PID: 5765
05-11 12:12:05.589  5765  5765 E AndroidRuntime: java.lang.IndexOutOfBoundsException: The list is empty.
05-11 12:12:05.589  5765  5765 E AndroidRuntime: 	at io.realm.RealmList.firstImpl(RealmList.java:449)
05-11 12:12:05.589  5765  5765 E AndroidRuntime: 	at io.realm.RealmList.first(RealmList.java:425)
05-11 12:12:05.589  5765  5765 E AndroidRuntime: 	at com.moez.QKSMS.feature.compose.ContactAdapter.onBindViewHolder(ContactAdapter.kt:65)
05-11 12:12:05.589  5765  5765 E AndroidRuntime: 	at com.moez.QKSMS.feature.compose.ContactAdapter.onBindViewHolder(ContactAdapter.kt:34)
05-11 12:12:05.589  5765  5765 E AndroidRuntime: 	at androidx.recyclerview.widget.RecyclerView$Adapter.onBindViewHolder(RecyclerView.java:6929)
05-11 12:12:05.589  5765  5765 E AndroidRuntime: 	at androidx.recyclerview.widget.RecyclerView$Adapter.bindViewHolder(RecyclerView.java:6971)
05-11 12:12:05.589  5765  5765 E AndroidRuntime: 	at androidx.recyclerview.widget.RecyclerView$Recycler.tryBindViewHolderByDeadline(RecyclerView.java:5891)
05-11 12:12:05.589  5765  5765 E AndroidRuntime: 	at androidx.recyclerview.widget.RecyclerView$Recycler.tryGetViewHolderForPositionByDeadline(RecyclerView.java:6158)
05-11 12:12:05.589  5765  5765 E AndroidRuntime: 	at androidx.recyclerview.widget.RecyclerView$Recycler.getViewForPosition(RecyclerView.java:5997)
05-11 12:12:05.589  5765  5765 E AndroidRuntime: 	at androidx.recyclerview.widget.RecyclerView$Recycler.getViewForPosition(RecyclerView.java:5993)
05-11 12:12:05.589  5765  5765 E AndroidRuntime: 	at androidx.recyclerview.widget.LinearLayoutManager$LayoutState.next(LinearLayoutManager.java:2227)
05-11 12:12:05.589  5765  5765 E AndroidRuntime: 	at androidx.recyclerview.widget.LinearLayoutManager.layoutChunk(LinearLayoutManager.java:1558)
05-11 12:12:05.589  5765  5765 E AndroidRuntime: 	at androidx.recyclerview.widget.LinearLayoutManager.fill(LinearLayoutManager.java:1518)
05-11 12:12:05.589  5765  5765 E AndroidRuntime: 	at androidx.recyclerview.widget.LinearLayoutManager.onLayoutChildren(LinearLayoutManager.java:613)
05-11 12:12:05.589  5765  5765 E AndroidRuntime: 	at androidx.recyclerview.widget.RecyclerView.dispatchLayoutStep2(RecyclerView.java:4029)
05-11 12:12:05.589  5765  5765 E AndroidRuntime: 	at androidx.recyclerview.widget.RecyclerView.dispatchLayout(RecyclerView.java:3746)
05-11 12:12:05.589  5765  5765 E AndroidRuntime: 	at androidx.recyclerview.widget.RecyclerView.onLayout(RecyclerView.java:4299)
05-11 12:12:05.589  5765  5765 E AndroidRuntime: 	at android.view.View.layout(View.java:19659)
05-11 12:12:05.589  5765  5765 E AndroidRuntime: 	at android.view.ViewGroup.layout(ViewGroup.java:6075)
05-11 12:12:05.589  5765  5765 E AndroidRuntime: 	at androidx.constraintlayout.widget.ConstraintLayout.onLayout(ConstraintLayout.java:1915)
05-11 12:12:05.589  5765  5765 E AndroidRuntime: 	at android.view.View.layout(View.java:19659)
05-11 12:12:05.589  5765  5765 E AndroidRuntime: 	at android.view.ViewGroup.layout(ViewGroup.java:6075)
05-11 12:12:05.589  5765  5765 E AndroidRuntime: 	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:323)
05-11 12:12:05.589  5765  5765 E AndroidRuntime: 	at android.widget.FrameLayout.onLayout(FrameLayout.java:261)
05-11 12:12:05.589  5765  5765 E AndroidRuntime: 	at android.view.View.layout(View.java:19659)
05-11 12:12:05.589  5765  5765 E AndroidRuntime: 	at android.view.ViewGroup.layout(ViewGroup.java:6075)
05-11 12:12:05.589  5765  5765 E AndroidRuntime: 	at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1791)
05-11 12:12:05.589  5765  5765 E AndroidRuntime: 	at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1635)
05-11 12:12:05.589  5765  5765 E AndroidRuntime: 	at android.widget.LinearLayout.onLayout(LinearLayout.java:1544)
05-11 12:12:05.589  5765  5765 E AndroidRuntime: 	at android.view.View.layout(View.java:19659)
05-11 12:12:05.589  5765  5765 E AndroidRuntime: 	at android.view.ViewGroup.layout(ViewGroup.java:6075)
05-11 12:12:05.589  5765  5765 E AndroidRuntime: 	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:323)
05-11 12:12:05.589  5765  5765 E AndroidRuntime: 	at android.widget.FrameLayout.onLayout(FrameLayout.java:261)
05-11 12:12:05.589  5765  5765 E AndroidRuntime: 	at android.view.View.layout(View.java:19659)
05-11 12:12:05.589  5765  5765 E AndroidRuntime: 	at android.view.ViewGroup.layout(ViewGroup.java:6075)
05-11 12:12:05.589  5765  5765 E AndroidRuntime: 	at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1791)
05-11 12:12:05.589  5765  5765 E AndroidRuntime: 	at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1635)
05-11 12:12:05.589  5765  5765 E AndroidRuntime: 	at android.widget.LinearLayout.onLayout(LinearLayout.java:1544)
05-11 12:12:05.589  5765  5765 E AndroidRuntime: 	at android.view.View.layout(View.java:19659)
05-11 12:12:05.589  5765  5765 E AndroidRuntime: 	at android.view.ViewGroup.layout(ViewGroup.java:6075)
05-11 12:12:05.589  5765  5765 E AndroidRuntime: 	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:323)
05-11 12:12:05.589  5765  5765 E AndroidRuntime: 	at android.widget.FrameLayout.onLayout(FrameLayout.java:261)
05-11 12:12:05.589  5765  5765 E AndroidRuntime: 	at com.android.internal.policy.DecorView.onLayout(DecorView.java:761)
05-11 12:12:05.589  5765  5765 E AndroidRuntime: 	at android.view.View.layout(View.java:19659)
05-11 12:12:05.589  5765  5765 E AndroidRuntime: 	at android.view.ViewGroup.layout(ViewGroup.java:6075)
05-11 12:12:05.589  5765  5765 E AndroidRuntime: 	at android.view.ViewRootImpl.performLayout(ViewRootImpl.java:2496)
05-11 12:12:05.589  5765  5765 E AndroidRuntime: 	at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:2212)
05-11 12:12:05.589  5765  5765 E AndroidRuntime: 	at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1392)
05-11 12:12:05.589  5765  5765 E AndroidRuntime: 	at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:6752)
05-11 12:12:05.589  5765  5765 E AndroidRuntime: 	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:911)
05-11 12:12:05.589  5765  5765 E AndroidRuntime: 	at android.view.Choreographer.doCallbacks(Choreographer.java:723)
05-11 12:12:05.589  5765  5765 E AndroidRuntime: 	at android.view.Choreographer.doFrame(Choreographer.java:658)
05-11 12:12:05.589  5765  5765 E AndroidRuntime: 	at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:897)
05-11 12:12:05.589  5765  5765 E AndroidRuntime: 	at android.os.Handler.handleCallback(Handler.java:790)
05-11 12:12:05.589  5765  5765 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:99)
05-11 12:12:05.590  5765  5765 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:164)
05-11 12:12:05.590  5765  5765 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:6494)
05-11 12:12:05.590  5765  5765 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
05-11 12:12:05.590  5765  5765 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:440)
05-11 12:12:05.590  5765  5765 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807)
05-11 12:12:05.596 19498 19518 I Finsky  : [411] iey.run(3): Stats for Executor: LightweightExecutor igj@9633751[Running, pool size = 3, active threads = 0, queued tasks = 0, completed tasks = 66]
05-11 12:12:05.638 19498 19518 I Finsky  : [411] iey.run(3): Stats for Executor: bgExecutor igj@3bf6bb6[Running, pool size = 4, active threads = 0, queued tasks = 0, completed tasks = 64]
05-11 12:12:05.938   725  3536 W ActivityManager:   Force finishing activity com.moez.QKSMS/.feature.compose.ComposeActivity
05-11 12:12:05.947   725  1079 I ActivityManager: Showing crash dialog for package com.moez.QKSMS u0
05-11 12:12:05.955   725  1071 W BroadcastQueue: Background execution not allowed: receiving Intent { act=android.intent.action.DROPBOX_ENTRY_ADDED flg=0x10 (has extras) } to com.google.android.gms/.stats.service.DropBoxEntryAddedReceiver
05-11 12:12:05.955   725  1071 W BroadcastQueue: Background execution not allowed: receiving Intent { act=android.intent.action.DROPBOX_ENTRY_ADDED flg=0x10 (has extras) } to com.google.android.gms/.chimera.GmsIntentOperationService$PersistentTrustedReceiver
05-11 12:12:06.005   725  8296 I OpenGLRenderer: Initialized EGL, version 1.4
05-11 12:12:06.005   725  8296 D OpenGLRenderer: Swap behavior 1
05-11 12:12:06.019   725  8296 W Adreno-EGL: <qeglDrvAPI_eglGetConfigAttrib:607>: EGL_BAD_ATTRIBUTE
05-11 12:12:06.022   725  8296 W Adreno-EGL: <qeglDrvAPI_eglGetConfigAttrib:607>: EGL_BAD_ATTRIBUTE
05-11 12:12:06.023   725  8296 D vndksupport: Loading /vendor/lib/hw/gralloc.msm8974.so from current namespace instead of sphal namespace.
05-11 12:12:06.439   725  1071 W ActivityManager: Activity pause timeout for ActivityRecord{dc0b5be u0 com.moez.QKSMS/.feature.compose.ComposeActivity t27 f}
```

**Comment by moezbhatti** [comment]
Fixed in ebf6014e703fb3f170a1de1c482a7fb030bc8b25 - included in the 3.6.5 release
